src/ui/public/vis/agg_configs.js

```diff
@@ -47,6 +47,23 @@ export default function AggConfigsFactory(Private) {
     }
   }
 
+  /**
+   * Data-by-data comparison of this Aggregation
+   * Ignores the non-array indexes
+   * @param aggConfigs an AggConfigs instance
+   */
+  AggConfigs.prototype.jsonDataEquals = function (aggConfigs) {
+    if (aggConfigs.length !== this.length) {
+      return false;
+    }
+    for (let i = 0; i < this.length; i += 1) {
+      if (!_.isEqual(aggConfigs[i].toJSON(),this[i].toJSON())) {
```

Comment:
@epixa wrt. serialization order.

This is more of a lodash issue. The isEqual method does not care about property order (https://jsfiddle.net/vamwrxmb/1/). So it is not fully equivalent to `JSON.stringify(foo)===JSON.stringify(bar);`, which orders them according to enumeration-order.

In both JSON (http://www.json.org/) and Ecmascript (http://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order) property order is explicitly undefined too. I gather that JS-engines enumerate over properties in insertion-order, but that apparently is not guaranteed in the specs.

Looking at  the `AggConfig`s implementation, a new object literal is returned which ensures consistent ordering for all intents and purposes. ie.  it does not enumerating over anything when serializing.

tl;dr; ordering is not likely to pose an issue here in this particular check. I would not add a unit test for this though, because  we would be testing peculiarities of the JS-runtime.

(nonetheless of course, the 'jsonDataEquals' method needs unit tests I agree).